addon/-private/rfc268-helpers.js

```diff
@@ -0,0 +1,36 @@
+//
+// This is a wrapper around `@ember/test-helpers` that we need for compatibility
+// reasons. Apps and addons aren't supposed to depend directly on
+// `@ember/test-helpers`, but just use the one that their version of
+// `ember-qunit` or `ember-mocha` provides. Since we want to support both
+// RFC268 tests (which will have `@ember/test-helpers` installed) and legacy
+// tests (which will have `ember-test-helpers` installed), we can't just import
+// `@ember/test-helpers` as an ES6 module because then even if we are running
+// legacy tests that don't actually use it, it will throw a module load error.
+// So, we use this module to check if it's present and if so, export all of the
+// helpers that we need, and if not, export a `getContext()` that will return
+// `null` so when picking an execution context we'll never pick RFC268, and
+// will never try to access any of the other helpers.
+//
+// Once we drop support for pre-RFC268 tests, we can delete this file and import
+// `@ember/test-helpers` directly.
+//
+import require from 'require';
+
+let helpers;
+
+if (require.has('@ember/test-helpers')) {
+  helpers = require('@ember/test-helpers');
+} else {
+  helpers = {
+    getContext() {
+      return null;
+    }
+  }
+}
```

Comment:
I think it would be helpful for this file to have something like this, to simplify the logic/limit the concerns of `-private/execution_context.js`.
```javascript
export function hasRfc268TestContext() {
  // `helpers.getContext()` returns:
  //  - `null`, if @ember/test-helpers is not available
  //  - _____?, if @ember/test-helpers is available but none of the `ember-qunit` 
  //    setupTest() methods has been called (e.g., `setupRenderingTest()`)
  //  - qunit test context, if @ember/test-helpers is available and one of the
  //    `ember-qunit` setupTest() methods has been called.
  let hasValidTestContext = !!helpers.getContext();

  if (!hasValidTestContext) {
    return false;
  }

  let hasExpectedTestHelpers = !!helpers.visit;  
  if (!hasExpectedTestHelpers) {
      throw new Error([
        'You are trying to use ember-cli-page-object with RFC232/RFC268 support',
        '(setupRenderingContext()/setupApplicationContext()) which requires at',
        'least ember-qunit@3.2.0 or ember-mocha@0.13.0-beta.3.'
      ]);
    }

  return true;
}
```

Then, in `-private/execution_context.js`:
```javascript
export function getExecutionContext(pageObjectNode) {
  let pageObjectContext = getContext(pageObjectNode);
  let context;

  if (pageObjectContext) {
    context = 'integration';
  } else if (hasRfc268TestContext()) {
    context = 'rfc268';
  } else {
    context = 'acceptance';
  }

  return new executioncontexts[context](pageObjectNode, pageObjectContext);
}
```

Reply:
Yeah, I guess I could do that, but there's not a really clean separation of concerns. The logic in `hasRfc268TestContext()` relies on the fact that we've already ruled out the integration context in order to be sure it's making the right compatibility/upgrade decisions for the user. We have to make `page.setContext()` force integration mode or we break compatibility with people that have started using `ember-cli-page-object` with RFC232 tests (which works fine on master).

So the logic is really a cascade of decisions:

1. Integration test or RFC232 test that is calling `page.setContext()`? => integration
1. RFC232 or RFC268 test with `visit` present? => rfc268
1. else => acceptance

and it seems to me that the importance of the ordering is harder to express when the decision logic isn't all in the same file.

I don't feel strongly about this, so I'm happy to move the code like you suggest, but my personal judgement is that the loss of locality takes away more from the readability than the (imperfect) separation of concerns adds. But say the word and I'll happily concede the point 😄

Reply:
Yeah, I say let's go ahead and make these changes. I think these explanatory comments and error handling belong in this file rather than `-private/execution_context.js`.

Reply:
Okay, pushed those changes (sorry it took a bit -- was on vacation and not connected for the long weekend)